mm2src/coins/lp_coins.rs

```diff
@@ -3570,27 +3590,149 @@ impl MmCoinStruct {
+                    return DexFee::WithBurn {
+                        fee_amount,
+                        burn_amount,
+                        burn_destination: DexFeeBurnDestination::BurnAccount,
+                    };
+                }
+            }
         }
+        DexFee::Standard(dex_fee)
+    }
+
+    /// Returns dex fee discount if KMD is traded
+    pub fn dex_fee_rate(base: &str, rel: &str) -> MmNumber {
+        let fee_discount_tickers: &[&str] = if var("MYCOIN_FEE_DISCOUNT").is_ok() {
+            &["KMD", "MYCOIN"] // used in tests
+        } else {
+            &["KMD"]
+        };
+        if fee_discount_tickers.contains(&base) || fee_discount_tickers.contains(&rel) {
+            // 1/777 - 10%
+            BigRational::new(9.into(), 7770.into()).into()
+        } else {
+            BigRational::new(1.into(), 777.into()).into()
+        }
+    }
+
+    fn calc_burn_amount_for_op_return(dex_fee: &MmNumber, min_tx_amount: &MmNumber) -> (MmNumber, MmNumber) {
+        // Drop the dex fee by 25%, which will be burned during the taker fee payment.
+        //
+        // This cut will be dropped before return if the final amount is less than
+        // the minimum transaction amount.
+
+        // Dex fee with 25% cut
+        let new_fee = dex_fee * &MmNumber::from(Self::BURN_CUT);
+
+        if &new_fee >= min_tx_amount {
+            // Use the max burn value, which is 25%.
+            let burn_amount = dex_fee - &new_fee;
+
+            // we don't care if burn_amount < dust as any amount can be sent to op_return
+            (new_fee, burn_amount)
+        } else {
+            // Burn only the exceed amount because fee after 25% cut is less
+            // than `min_tx_amount`.
+            let burn_amount = dex_fee - min_tx_amount;
+
+            (min_tx_amount.clone(), burn_amount)
+        }
+    }
+
+    fn calc_burn_amount_for_burn_account(dex_fee: &MmNumber, min_tx_amount: &MmNumber) -> (MmNumber, MmNumber) {
+        // Dex fee with 25% cut
+        let new_fee = dex_fee * &MmNumber::from(Self::BURN_CUT);
+
+        if &new_fee >= min_tx_amount {
+            // Use the max burn value, which is 25%.
+            let burn_amount = dex_fee - &new_fee;
+
+            // as burn_amount is actually sent to an account it should not be dust
+            if &burn_amount >= min_tx_amount {
+                return (new_fee, burn_amount);
+            }
+        } else {
+            // Burn only the exceed amount because fee after 25% cut is less than `min_tx_amount`.
+            let burn_amount = dex_fee - min_tx_amount;
+
+            // burn_amount should not be dust
+            if &burn_amount >= min_tx_amount {
+                // actually currently burn_amount (25%) < new_fee (75%) so this never happens.
+                // added for a case if 25/75 will ever change
+                return (min_tx_amount.clone(), burn_amount);
+            }
+        }
+        (dex_fee.clone(), 0.into())
```

Comment:
You can avoid using if-else here by simplifying the code to two separate if checks, as both conditions result in early returns.

Please re-check that I didnt miss the calculation logic
```rust
    fn calc_burn_amount_for_burn_account(dex_fee: &MmNumber, min_tx_amount: &MmNumber) -> (MmNumber, MmNumber) {
        // Dex fee with 25% cut
        let new_fee = dex_fee * &MmNumber::from(Self::BURN_CUT);

        let burn_amount = dex_fee - &new_fee;
        if &new_fee >= min_tx_amount && &burn_amount >= min_tx_amount {
            // Use the max burn value, which is 25%. Ensure burn_amount is not dust
            return (new_fee, burn_amount);
        }

        // Burn only the exceed amount because fee after 25% cut is less than `min_tx_amount`.
        let burn_amount = dex_fee - min_tx_amount;
        // burn_amount should not be dust
        if &burn_amount >= min_tx_amount {
            // actually currently burn_amount (25%) < new_fee (75%) so this never happens. Added for a case if 25/75 will ever change
            return (min_tx_amount.clone(), burn_amount);
        }

        // Default case where burn_amount is considered dust
        (dex_fee.clone(), 0.into())
    }
```

Reply:
logic will be different then.
The idea was if (hypothetically, if proportion is ever changed so and burn > dex fee) new_fee is dust and burn_amount is not dust, then make new_fee as min_tx_amount and recalc burn_amount

Reply:
fixed in [7a2399c](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2112/commits/7a2399c795d45ebc6fb8017cfacdd245fc9ff715)